src/kbmod/analysis/plotting.py

```diff
@@ -0,0 +1,382 @@
+            xoffset = txt_x_offset.to(u.deg).value
+            yoffset = txt_y_offset.to(u.deg).value
+            # we need to move diagonally to the right and down to center the text
+            x, y = pt.ra.deg + xoffset, pt.dec.deg + yoffset
+            ax.text(x, y, hdu.header.get("EXTNAME", None), clip_on=True)
+
+    # ax = plot_footprints(ax, wcss)
+    return ax
+
+
+def plot_cutouts(axes, cutouts, remove_extra_axes=True):
+    """Plots cutouts (images) onto given axes.
+
+    The number of axes must be equal to or greater
+    than the number of cutouts.
+
+    Parameters
+    ----------
+    ax : `list[matplotlib.pyplot.Axes]`
+        Axes.
+    cutouts : `list`, `np.array` or `astropy.ndutils.Cutout2D`
+        Collection of numpy arrays or ``Cutout2D`` objects
+        to plot.
+    remove_extra_axes : `bool`, optional
+         When `True` (default), the axes that would be
+         left empty are removed from the plot.
+
+    Raises
+    -------
+    ValueError - When number of given axes is less than
+    the number of given cutouts.
+    """
+    nplots = len(cutouts)
+
+    axs = axes.ravel()
+    naxes = len(axs)
+
+    if naxes < nplots:
+        raise ValueError(f"N axes ({len(axs)}) doesn't match N plots ({nplots}).")
+
+    for ax, cutout in zip(axs, cutouts):
+        img = cutout.data if isinstance(cutout, Cutout2D) else cutout
+        norm = ImageNormalize(img, interval=ZScaleInterval(), stretch=AsinhStretch())
+        im = ax.imshow(cutout.data, norm=norm)
+        ax.set_aspect("equal")
+        ax.axvline(cutout.shape[0] / 2, c="red", lw=0.25)
+        ax.axhline(cutout.shape[1] / 2, c="red", lw=0.25)
+
+    if remove_extra_axes and naxes > nplots:
+        for ax in axs[nplots - naxes :]:
+            ax.remove()
+
+    return axes
+
+
+def plot_img(img, ax=None, figure=None, norm=True, title=None):
+    """Plots an image on an axis and figure.
+
+    If no axis is given creates a new figure. Draws a crosshair at the
+    center of the image. Must provide both axis and figure; figure is
+    required so a colorbar could be attached to it.
+
+    Parameters
+    ----------
+    img : `np.array`
+        Image array
+    ax : `matplotlib.pyplot.Axes` or `None`
+        Axes, `None` by default.
+    figure : `matplotlib.pyplot.Figure` or `None`
+        Figure, `None` by default.
+    norm: `bool`, optional
+        Normalize the image using Astropy's `ImageNormalize`
+        using `ZScaleInterval` and `AsinhStretch`. `True` by
+        default.
+    title : `str` or None, optional
+        Title of the plot.
+
+    Returns
+    -------
+    figure : `matplotlib.pyplot.Figure`
+        Modified Figure.
+    ax : `matplotlib.pyplot.Axes`
+        Modified Axes.
+    """
+    if ax is None and figure is None:
+        fig, ax = plt.subplots(figsize=(15, 10))
+    elif ax is not None and figure is not None:
+        raise ValueError("Provide both figure and axis, or provide none.")
```

Comment:
I don't under stand this error. The phrasing makes it sound like you need to provide both axis and figure, but it only triggers when both are provided. Should it say "Provide either a figure or axis"?

Reply:
Whoops, that should've been an `or`. Thanks